spec/stack_master/template_compilers/cfndsl_spec.rb

```diff
@@ -18,5 +18,27 @@ def compile
         expect(JSON.parse(compile)).to eq(JSON.parse(valid_compiled_json))
       end
     end
+
+    context 'with compile time parameters' do
+      let(:template_file_path) { 'spec/fixtures/templates/rb/cfndsl/sample-ctp.rb' }
+      let(:valid_compiled_json_path) { 'spec/fixtures/templates/rb/cfndsl/sample-ctp.json' }
+
+      it 'produces valid JSON' do
+        valid_compiled_json = File.read(valid_compiled_json_path)
+        expect(JSON.parse(compile)).to eq(JSON.parse(valid_compiled_json))
+      end
+
+      context 'compiling multiple times' do
+        let(:compile_time_parameters) { {'InstanceType' => 't2.medium', 'DisableApiTermination' => 'true'} }
+        let(:template_file_path) { 'spec/fixtures/templates/rb/cfndsl/sample-ctp-repeated.rb' }
+
+        it 'does not leak compile time params across invocations' do
+          #expect(JSON.parse(compile)["Resources"]["MyInstance"]["Properties"]["DisableApiTermination"]).to eq(compile_time_parameters['DisableApiTermination'])
```

Comment:
What's the go with this line commented out? I would've thought you'd need to call compile here to ensure the clear actually works.

Reply:
Sorry, debug leftover!